### PR TITLE
Modification to `WaveformContainer` and `ChargeContainer` for SPE-WT

### PR DESCRIPTION
**Pull request by sona-patel**
This PR modifies the `WaveformContainer` and `ChargeContainer` (mainly `load_wfs` and `write` function) to read and store waveform given the list of event_id for splitting.

### CODE REVIEW

**Comment by jlenain** [comment]
Thanks, @sona-patel !
Is this PR still in the draft stage ? The title seems to indicate so.

File: src/nectarchain/calibration/container/charge.py [hunk 1]
```diff
@@ -124,140 +126,212 @@ def __init__(self,charge_hg,charge_lg,peak_hg,peak_lg,run_number,pixels_id,neven
...
         chargeContainer.event_type = waveformContainer.event_type
         chargeContainer.event_id = waveformContainer.event_id
         chargeContainer.trig_pattern_all = waveformContainer.trig_pattern_all
 
-        return chargeContainer 
+        return chargeContainer
 
-
-    def write(self,path : Path,**kwargs) : 
+    def write(self, path: Path, bloc, start, end, **kwargs):
```
Comment (jlenain) [inline src/nectarchain/calibration/container/charge.py:186]: I think `bloc` (`block` ?), `start` and `end` should have default values such that a whole run is processed and written if no value are provided for these arguments.

File: src/nectarchain/calibration/container/charge.py [hunk 2]
```diff
@@ -124,140 +126,212 @@ def __init__(self,charge_hg,charge_lg,peak_hg,peak_lg,run_number,pixels_id,neven
...
+        hdr["NPIXELS"] = self.npixels
+        hdr["COMMENT"] = (
+            f"The charge containeur for run {self._run_number} with {self._method} "
+            f"method : primary is the pixels id, then you can find HG charge, "
+            f"LG charge, HG peak and LG peak, 2 last HDU are composed of event "
+            f"properties and trigger patern"
+        )
+
+        primary_hdu = fits.PrimaryHDU(self.pixels_id, header=hdr)
+        charge_hg_hdu = fits.ImageHDU(self.charge_hg[start:end], name="HG charge")
```
Comment (jlenain) [inline src/nectarchain/calibration/container/charge.py:227]: For instance here, the default behavior should allow for the full `charge_hg` attribute to be stored and written to disk.

File: src/nectarchain/calibration/container/waveforms.py [hunk 3]
```diff
@@ -136,138 +157,196 @@ def check_events(self):
...
+                    f"trigger type"
+                )
+            n_events += 1
         return n_events
 
-
-
-    def load_wfs(self,compute_trigger_patern = False):
-        """mathod to extract waveforms data from the EventSource 
+    def load_wfs(self, start, end, compute_trigger_patern=False):
```
Comment (jlenain) [inline src/nectarchain/calibration/container/waveforms.py:172]: A similar comment here, I think `start` and `end` should have default values such that the all events should be loaded.

File: src/nectarchain/calibration/container/waveforms.py [hunk 4]
```diff
@@ -136,138 +157,196 @@ def check_events(self):
...
-            self.ucts_timestamp[i]=event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_timestamp
-            self.event_type[i]=event.trigger.event_type.value
-            self.ucts_busy_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_busy_counter
-            self.ucts_event_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_event_counter
+        wfs_hg_tmp = np.zeros((self.npixels, self.nsamples), dtype=np.uint16)
+        wfs_lg_tmp = np.zeros((self.npixels, self.nsamples), dtype=np.uint16)
 
+        # n_traited_events = 0
+        event_start, event_end = start, end
+        print(event_start, event_end)
```
Comment (jlenain) [inline src/nectarchain/calibration/container/waveforms.py:187]: Plain prints should be avoided, to the benefit of the logging system.

File: src/nectarchain/calibration/container/waveforms.py [hunk 5]
```diff
@@ -136,138 +157,196 @@ def check_events(self):
...
 
-        #if compute_trigger_patern and np.max(self.trig_pattern) == 0:
+        # if compute_trigger_patern and np.max(self.trig_pattern) == 0:
         #    self.compute_trigger_patern()
 
-
-    def write(self,path : str, **kwargs) : 
-        """method to write in an output FITS file the WaveformsContainer. Two files are created, one FITS representing the data
-        and one HDF5 file representing the subarray configuration
+    def write(self, start, end, bloc, path: str, **kwargs):
```
Comment (jlenain) [inline src/nectarchain/calibration/container/waveforms.py:227]: Here also, `start`, `end`, and `bloc` (`block` ?) should default to the whole run.

**Review by jlenain — CHANGES_REQUESTED** [review]
Thanks for this, @sona-patel . I think the class behaviors should default to loading/writting whole runs. In such a case, default values should be provided in the loading/writting methods.

**Comment by sona-patel** [comment]
> Thanks, @sona-patel ! Is this PR still in the draft stage ? The title seems to indicate so.

Thanks for reviewing @jlenain!
Yes, I kept it as a draft, since I also want to add the user_scripts using this case. What do you suggest? Should I put it along with this PR or in a separate PR?

File: src/nectarchain/calibration/container/waveforms.py (hunk 4 above)
Comment (sona-patel) [reply src/nectarchain/calibration/container/waveforms.py:187]: Agreed! Its remnant from the notebook 😀!

**Review by jlenain — APPROVED** [review]

**Pull request merged**